### PR TITLE
Document the Wattsi language

### PR DESCRIPTION
**Pull request by domenic**
@clelland and @yoavweiss have recently had to struggle through this, and I'd love their review or thoughts. Sorry this wasn't in place already.

### CODE REVIEW

File: Syntax.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+|Multipage          |w-split |w-nosplit |
+|Developer's edition|w-dev   |w-nodev   |
+|Commit snapshots   |w-snap  |w-nosnap  |
+|Review drafts      |w-review|w-noreview|
+
+The most important of these for day-to-day work is **w-nodev**, which excludes content (such as details only interesting to web browser implementers) from the developer's edition.
+
+## Macros
+
+Wattsi has a couple of macros used to insert common text fragments:
```
[the remark this answers is not in the document]
Comment (clelland) [reply Syntax.md]: Actually, these macros are extremely specific -- even an extra space will cause it not to be recognized. Only an exact byte match for `<!-- NON-NORMATIVE SECTION -->` seems to trigger it.

File: Syntax.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+* `<!-- NON-NORMATIVE SECTION -->` inserts a "This section is non-normative." notice. ([Example](https://html.spec.whatwg.org/multipage/introduction.html#suggested-reading))
+* `<!--INSERT FINGERPRINT-->` inserts a fingerprint image to indicate a feature that may cause fingerprinting. (See the "[Privacy concerns](https://html.spec.whatwg.org/multipage/introduction.html#fingerprint)" section)
```
Comment (clelland) [inline Syntax.md]: Also note that there are no spaces around this one.

The other macros are `<!--smalltoc-->` and `<!--toc-->`, which generate the short and full tables of contents, respectively. That's probably not of interest to most contributors to HTML, but if the point is to document the Wattsi syntax, they may deserve a mention.

**Comment by domenic** [comment]
Thanks @clelland, updated with your suggestions!

It turns out the attribute index rejiggering is done by https://github.com/whatwg/html-build/blob/master/.pre-process-annotate-attributes.pl, another part of the pipeline :-/. I still agree we should document it, and link to the docs from Contributing.md, but probably not in this file.

**Comment by clelland** [comment]
This LGTM -- agreed, that if the magic happens elsewhere, we should document it there, in whatwg/html-build somewhere.

File: Syntax.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+```html
+<p>Each <span>custom element definition</span> has a <dfn
+data-x="concept-custom-element-definition-name">name</dfn>.</p>
+
+<p>In general, algorithms look up custom elements by any of <span
+data-x="concept-custom-element-definition-name">name</span>, ...</p>
+```
+
+Note that the choice of `"concept-custom-element-definition-name"` was arbitrary; any unique string would suffice. However, since the ID for a `<dfn>` (and thus the URL fragment for any links to it) is derived from its canonical definition string, we try to choose ones that look nice.
```
Comment (yoavweiss) [inline Syntax.md]: Maybe `s/look nice/look nice and are meaningful/` or something like that?

File: Syntax.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+```html
+<p>Each <code>DOMStringList</code> object has an associated <span>list</span>.</p>
+
+<p><span data-x="list append">Append</span> <var>serializedEntry</var> to
+<var>serialized</var>.[[SetData]].</p>
+```
+
+### Exporting definitions
+
+Definitions that are meant to be used by other specifications need to be exported using the [Bikeshed definition data model](https://tabatkins.github.io/bikeshed/#dfn-contract). In practice, this means adding `data-export=""` to the `<dfn>`, and if appropriate, `data-dfn-type=""`, `data-lt=""`, or `data-dfn-for=""` attributes. Some examples:
```
Comment (yoavweiss) [inline Syntax.md]: Maybe link to the definitions directly?
https://tabatkins.github.io/bikeshed/#local-lt
https://tabatkins.github.io/bikeshed/#dfn-types
https://tabatkins.github.io/bikeshed/#dfn-for

**Comment by domenic** [comment]
Yay! @sideshowbarker do you want to do a final pass, since you're probably our best Wattsi expert? Feel free to decline though if you're busy; we can merge now and tweak later.

**Pull request merged**